### PR TITLE
Install Prettier

### LINKED ISSUE #1347: Add prettier in CI
For both JavaScript and Ruby

### PR DESCRIPTION
**Pull request by wkillerud**
- Install Prettier
- Run Prettier on the project with the default configuration
- `git config --local blame.ignoreRevsFile .git-blame-ignore-revs` to ignore the Prettifying commit in blames.
- Add husky and lint-staged to run Prettier as a pre-commit hook so we don't fail on CI. `yarn install` to set up.

ESLint is also something I usually add with the default recommended settings and run on pre-commit + CI. Want to add that as well?

Fixes #1347

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,72 +1,79 @@
...
+  gem "rails-controller-testing"
+  gem "simplecov", require: false
+  gem "simplecov-lcov", require: false
+  gem "webmock"
 end
+
+gem "prettier_print", "~> 1.2"
+gem "syntax_tree", "~> 5.2"
+gem "syntax_tree-haml", "~> 3.0"
+gem "syntax_tree-rbs", "~> 0.5.1"
```
Comment (ujh) [inline Gemfile:79]: Those should go into the `group :development do ... end` block. Those are the Ruby equivalent of dev dependencies that do not need to installed in production.

**Review by ujh — APPROVED** [review]
One minor tweak, otherwise a 👍🏻 from me.

I do like the idea of eslint, but a separate PR would probably be best for that.

**Comment by ujh** [comment]
Oh, and a prettier config would be great, too. When my editor runs prettier on the ruby files the format is slightly different. I think having an explicit config in the project would solve that.

**Comment by wkillerud** [comment]
Done. I'll make a separate issue for ESLint :)

**Pull request merged**